---
title: "Ch1Heat_virome"
output: html_document
date: "2023-11-15"
---

# UPLOAD AND PREP DATA
```{r}
# LOAD PACKAGES
library(tidyverse)
library(vegan)
library(lubridate)
library(dplyr)
library(biobroom)
library(DESeq2)
library(factoextra)
library(gridExtra)
library(cowplot)
library(ComplexUpset)

# SET WORKING DIRECTORY
setwd("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures")

# LOAD ABUNDANCE DATA FOR ENTIRE DATASET
otu=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/all.75.tmean.tsv", delim = "\t", col_names = TRUE)

# PREP DATA FRAME
otu <- col_to_row(otu, "Contig", "contignames")
# edit column names 
colnames(otu) <- gsub(pattern=".vib.sI.Trimmed.Mean", replacement = "", x=colnames(otu))
# remove two samples that are not necessary
otu$N1_S61_L004 <- NULL
otu$N2_S62_L004 <- NULL
write.csv(otu, "vOTU_abundance_table.csv")

# LOAD METADATA FOR ENTIRE DATASET
metadata=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/heat_metadata.txt", delim = "\t", col_names = TRUE)
colnames(metadata)[colnames(metadata) == "Treatment"] <- "Experiment"
metadata <- metadata %>%
  mutate(Treatment = case_when(
    Temp == "Control" ~ "Field",
    Temp == "10C" ~ "Control",
    Temp == "30C" ~ "30ºC",
    Temp == "60C" ~ "60ºC", 
    Temp == "90C" ~ "90ºC")) %>%
  mutate(TempGroup = case_when(
    Treatment == "Field" ~ "Field/Control/30ºC", 
    Treatment == "Control" ~ "Field/Control/30ºC", 
    Treatment == "30ºC" ~ "Field/Control/30ºC", 
    Treatment == "60ºC" ~ "60ºC", 
    Treatment == "90ºC" ~ "90ºC")) %>%
  mutate(Virome = case_when(
    DNase == "NoDNase" ~ "NonDNase",
    DNase == "DNase" ~ "DNase"))
write.csv(metadata, "heatviromemetadata.csv", row.names = FALSE)

# LOAD COUNT TABLE
otu.count=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/all.count.tsv", delim = "\t", col_names = TRUE)
otu.count <- col_to_row(otu.count, "Contig", "contignames")
colnames(otu.count) <- gsub(pattern=".vib.sI.Read.Count", replacement = "", x=colnames(otu.count))
otu.count=otu.count%>%
  select(-N1_S61_L004, -N2_S62_L004)
```
# SUBSET DATA: Abundance(Remove Singletons & Relativize), Metadata, & Count)
# HORIZON
```{r}
# A horizon
a.otu <- otu[,metadata$Horizon=="A"] %>%
  rmv_sngl()
a.otu.rel <- relativize(a.otu)
colSums(a.otu.rel)
a.map <- subset(metadata, Horizon=="A")
a.count <- otu.count[,metadata$Horizon=="A"]
a.count <- rmv_sngl(a.count)

# O horizon
o.otu <- otu[,metadata$Horizon=="O"] %>%
  rmv_sngl()
o.otu.rel <- relativize(o.otu)
colSums(o.otu.rel)
o.map <- subset(metadata, Horizon=="O")
o.count <- otu.count[,metadata$Horizon=="O"]
o.count <- rmv_sngl(o.count)
```
# DNASE TREATMENT
```{r}
# DNase
d.otu <- otu[,metadata$DNase=="DNase"] %>%
  rmv_sngl()
d.otu.rel <- relativize(d.otu)
colSums(d.otu.rel)
d.map <- subset(metadata, DNase=="DNase")
d.count <- otu.count[,metadata$DNase=="DNase"]
d.count <- rmv_sngl(d.count)

# No DNase
n.otu <- otu[,metadata$DNase=="Untreated"] %>%
  rmv_sngl()
n.otu.rel <- relativize(n.otu)
colSums(n.otu.rel)
n.map <- subset(metadata, DNase=="Untreated")
n.count <- otu.count[,metadata$DNase=="Untreated"]
n.count <- rmv_sngl(n.count)
```
# HORIZON & DNASE TREATMENT
```{r}
# A horizon + DNase
ad.otu <- otu[,metadata$Horizon=="A"&metadata$Virome=="DNase"] %>%
  rmv_sngl()
ad.otu.rel <- relativize(ad.otu)
colSums(ad.otu.rel)
ad.map <- subset(metadata, Horizon=="A"&metadata$Virome=="DNase")
ad.count <- otu.count[,metadata$Horizon=="A"&metadata$Virome=="DNase"]
ad.count <- rmv_sngl(ad.count)

# A horizon + No DNase
an.otu <- otu[,metadata$Horizon=="A"&metadata$Virome=="NonDNase"] %>%
  rmv_sngl()
an.otu.rel <- relativize(an.otu)
colSums(an.otu.rel)
an.map <- subset(metadata, Horizon=="A"&metadata$Virome=="NonDNase")
an.count <- otu.count[,metadata$Horizon=="A"&metadata$Virome=="NonDNase"]
an.count <- rmv_sngl(an.count)

# O horizon + DNase
od.otu <- otu[,metadata$Horizon=="O"&metadata$Virome=="DNase"] %>%
  rmv_sngl()
od.otu.rel <- relativize(od.otu)
colSums(od.otu.rel)
od.map <- subset(metadata, Horizon=="O"&metadata$Virome=="DNase")
od.count <- otu.count[,metadata$Horizon=="O"&metadata$Virome=="DNase"]
od.count <- rmv_sngl(od.count)

# O horizon + No DNase
on.otu <- otu[,metadata$Horizon=="O"&metadata$Virome=="NonDNase"] %>%
  rmv_sngl()
on.otu.rel <- relativize(on.otu)
colSums(on.otu.rel)
on.map <- subset(metadata, Horizon=="O"&metadata$Virome=="NonDNase")
on.count <- otu.count[,metadata$Horizon=="O"&metadata$Virome=="NonDNase"]
on.count <- rmv_sngl(on.count)
```
# REMOVING SINGLETONS & RELATIVIZE (if analyzing all OTU together)
```{r}
dfname <- "otu"
df <- get(dfname)
otu <- rmv_sngl(df)
otu.rel <- relativize(otu)
colSums(otu.rel)
```
# DNA Yields
```{r}
yields=read_delim("DNAyield.txt", delim = "\t", col_names = TRUE)
yields <- data.frame(yields)
yields <- yields %>%
  mutate(Treatment = case_when(
    Temperature == "Control" ~ "Field",
    Temperature == "10" ~ "Control",
    Temperature == "30" ~ "30ºC",
    Temperature == "60" ~ "60ºC", 
    Temperature == "90" ~ "90ºC")) %>%
  mutate(TempGroup = case_when(
    Treatment == "Field" ~ "Field/Control/30ºC", 
    Treatment == "Control" ~ "Field/Control/30ºC", 
    Treatment == "30ºC" ~ "Field/Control/30ºC", 
    Treatment == "60ºC" ~ "60ºC", 
    Treatment == "90ºC" ~ "90ºC")) %>%
  mutate(DNase = case_when(
    DNase == "NoDNase" ~ "NonDnase",
    DNase == "DNase" ~ "DNase"))

DNAyield <- yields %>%
  ggplot(aes(x = factor(Treatment, level=temp_order), y = DNAyield, fill = factor(Horizon, level=horizon_order))) +
  geom_boxplot(outlier.shape=NA) +
  geom_point(position=position_jitterdodge()) +
  xlab("Temperature (°C)") +
  ylab("DNA Yield (ng)") +
  theme_bw() +
  theme(axis.title = element_text(size = 20)) +
  theme(legend.position = "bottom", legend.text = element_text(size=20), legend.title = element_text(size=20)) +
  theme(axis.text = element_text(size = 16)) +
  scale_fill_manual(name = "Horizon", 
                    values = horizon_colors) +
  facet_wrap(~ DNase)

DNAyield + theme(strip.text = element_text(size=20))

ggsave("Heat_DNAyield.pdf", width = 200, height = 120, units = "mm", dpi = 500)

# stats on DNA yield
yields.stats <- yields
yields_od <- yields.stats %>%
  filter(Horizon == "O"&DNase == "DNase")
kruskal.test(DNAyield ~ Treatment, yields_od)

yields_ad <- yields.stats %>%
  filter(Horizon == "A"&DNase == "DNase")
kruskal.test(DNAyield ~ Treatment, yields_ad)

yields_on <- yields.stats %>%
  filter(Horizon == "O"&DNase == "NonDnase")
kruskal.test(DNAyield ~ Treatment, yields_on)

yields_an <- yields.stats %>%
  filter(Horizon == "A"&DNase == "NonDnase")
kruskal.test(DNAyield ~ Treatment, yields_an)
```

# PCOA
```{r}
otu.dist <- distance(otu.rel)
otu.pcoa <- pcoa(otu.dist)
otu.pcoa.points <- pcoa.points(otu.pcoa)
otu.map <- pcoa.map(otu.pcoa.points, metadata)
variance(otu.pcoa, 1)
variance(otu.pcoa, 2)
pcoa.plot.temp(otu.map, x_col = pcoa1, y_col = pcoa2, color_col= Temp, shape_col = DNase, x_label = "74.76% Variance Explained", y_label = "8.73% Variance Explained", title = "All Data", filename = "all_heat_pcoa.png")
horizon_pcoa <- pcoa.plot.horizon(otu.map, x_col = pcoa1, y_col = pcoa2, color_col= Horizon, shape_col = Virome, x_label = "74.76% Variance Explained", y_label = "8.73% Variance Explained", title = NULL, filename = "all_heat_pcoa_horizon.pdf")

horizon_pcoa
saveRDS(horizon_pcoa, "horizon_pcoa.rds")

a.otu.dist <- distance(a.otu.rel)
a.otu.pcoa <- pcoa(a.otu.dist)
a.otu.pcoa.points <- pcoa.points(a.otu.pcoa)
a.otu.map <- pcoa.map(a.otu.pcoa.points, a.map)
variance(a.otu.pcoa, 1)
variance(a.otu.pcoa, 2)
a_temp_pcoa <- pcoa.plot.temp(a.otu.map, x_col = pcoa1, y_col = pcoa2, color_col= Treatment, shape_col = Virome, x_label = "55.36% Variance Explained", y_label = "17.80% Variance Explained", title = "A Horizon", filename = "a_heat_pcoa.pdf")
pcoa.plot.tempgroup(a.otu.map, x_col = pcoa1, y_col = pcoa2, color_col= TempGroup, shape_col = DNase, x_label = "55.18% Variance Explained", y_label = "17.69% Variance Explained", title = "A Horizon", filename = "a_group_heat_pcoa.png")

saveRDS(a_temp_pcoa, "a_temp_pcoa.rds")

o.otu.dist <- distance(o.otu.rel)
o.otu.pcoa <- pcoa(o.otu.dist)
o.otu.pcoa.points <- pcoa.points(o.otu.pcoa)
o.otu.map <- pcoa.map(o.otu.pcoa.points, o.map)
variance(o.otu.pcoa, 1)
variance(o.otu.pcoa, 2)
o_temp_pcoa <- pcoa.plot.temp(o.otu.map, x_col = pcoa1, y_col = pcoa2, color_col= Treatment, shape_col = Virome, x_label = "53.26% Variance Explained", y_label = "16.58% Variance Explained", title = "O Horizon", filename = "o_heat_pcoa.pdf")
pcoa.plot.tempgroup(o.otu.map, x_col = pcoa1, y_col = pcoa2, color_col= TempGroup, shape_col = DNase, x_label = "53.26% Variance Explained", y_label = "16.58% Variance Explained", title = "O Horizon", filename = "o_group_heat_pcoa.png")

saveRDS(o_temp_pcoa, "o_temp_pcoa.rds")

ad.otu.dist <- distance(ad.otu.rel)
ad.otu.pcoa <- pcoa(ad.otu.dist)
ad.otu.pcoa.points <- pcoa.points(ad.otu.pcoa)
ad.otu.map <- pcoa.map(ad.otu.pcoa.points, ad.map)
variance(ad.otu.pcoa, 1)
variance(ad.otu.pcoa, 2)
pcoa.plot.temp(ad.otu.map, x_col = pcoa1, y_col = pcoa2, color_col= Treatment, shape_col = NULL, x_label = "49.91% Variance Explained", y_label = "13.67% Variance Explained", title = "A Horizon, DNase Treated", filename = "ad_heat_pcoa.pdf")

od.otu.dist <- distance(od.otu.rel)
od.otu.pcoa <- pcoa(od.otu.dist)
od.otu.pcoa.points <- pcoa.points(od.otu.pcoa)
od.otu.map <- pcoa.map(od.otu.pcoa.points, od.map)
variance(od.otu.pcoa, 1)
variance(od.otu.pcoa, 2)
pcoa.plot.temp(od.otu.map, x_col = pcoa1, y_col = pcoa2, color_col= Treatment, shape_col = NULL, x_label = "53.64% Variance Explained", y_label = "15.31% Variance Explained", title = "O Horizon, DNase Treated", filename = "od_heat_pcoa.pdf")

an.otu.dist <- distance(an.otu.rel)
an.otu.pcoa <- pcoa(an.otu.dist)
an.otu.pcoa.points <- pcoa.points(an.otu.pcoa)
an.otu.map <- pcoa.map(an.otu.pcoa.points, an.map)
variance(an.otu.pcoa, 1)
variance(an.otu.pcoa, 2)
pcoa.plot.temp(an.otu.map, x_col = pcoa1, y_col = pcoa2, color_col= Treatment, shape_col = NULL, x_label = "49.01% Variance Explained", y_label = "16.77% Variance Explained", title = "A Horizon, Untreated", filename = "an_heat_pcoa.pdf")

on.otu.dist <- distance(on.otu.rel)
on.otu.pcoa <- pcoa(on.otu.dist)
on.otu.pcoa.points <- pcoa.points(on.otu.pcoa)
on.otu.map <- pcoa.map(on.otu.pcoa.points, on.map)
variance(on.otu.pcoa, 1)
variance(on.otu.pcoa, 2)
pcoa.plot.temp(on.otu.map, x_col = pcoa1, y_col = pcoa2, color_col= Treatment, shape_col = NULL, x_label = "68.28% Variance Explained", y_label = "12.05% Variance Explained", title = "O Horizon, Untreated", filename = "on_heat_pcoa.pdf")

```
# PCOA STATS
```{r}
# run statistics
adonis(otu.dist
~otu.map$'Horizon')
adonis(otu.dist
~otu.map$'DNase')
adonis(otu.dist
~otu.map$'Temp')
adonis(a.otu.dist
~a.otu.map$'Temp')
adonis(o.otu.dist
~o.otu.map$'Temp')

adonis(ad.otu.dist
~ad.otu.map$'Temp')
adonis(an.otu.dist
~an.otu.map$'Temp')
adonis(od.otu.dist
~od.otu.map$'Temp')
adonis(on.otu.dist
~on.otu.map$'Temp')

adonis(a.otu.dist
~a.otu.map$'TempGroup')
adonis(o.otu.dist
~o.otu.map$'TempGroup')
```
# UPSET PLOTS
```{r}
# upset plot for DNase Samples
map <- d.map

amap <- map %>%
  filter(Horizon=="A")
omap <- map %>%
  filter(Horizon=="O")

d.upset <- d.otu.rel

a_horizon <- d.upset[,colnames(d.upset)%in%amap$ID]
o_horizon <- d.upset[,colnames(d.upset)%in%omap$ID]

a_horizon_pa <- rowSums(a_horizon)>0  
o_horizon_pa <- rowSums(o_horizon) >0
d.upsetdata <- data.frame(a_horizon=a_horizon_pa, o_horizon=o_horizon_pa)

horizon_upset <- upset(data = d.upsetdata, intersect = c("a_horizon","o_horizon") ,
      name="vOTU Presence in DNase Viromes by Horizon",
      min_size = 0,
      width_ratio = 0.125,
      themes=upset_default_themes(text=element_text(size = 20)),
      sort_sets=FALSE,
      set_sizes=(upset_set_size()+theme(axis.text.x = element_text(angle=90))),
      stripes=upset_stripes(colors=c('darkorange', 'brown'))) #+
    #labs(title = "vOTU presence by Horizon")


saveRDS(horizon_upset, "horizon_upset.rds")
ggsave("d_upset.pdf", width = 200, height = 220, units = "mm", dpi = 500)
```
```{r}
# A horizon, untreated
# upset plot for A horizon, no DNase
map <- an.map

fieldmap <- map %>%
  filter(Treatment=="Field")
controlmap <- map %>%
  filter(Treatment=="Control")
thirtymap <- map %>%
  filter(Treatment=="30ºC")
sixtymap <- map %>%
  filter(Treatment=="60ºC")
ninetymap <- map %>%
  filter(Treatment=="90ºC")

an.upset <- an.otu.rel

field <- an.upset[,colnames(an.upset)%in%fieldmap$ID]
control <- an.upset[,colnames(an.upset)%in%controlmap$ID]
thirty <- an.upset[,colnames(an.upset)%in%thirtymap$ID]
sixty <- an.upset[,colnames(an.upset)%in%sixtymap$ID]
ninety <- an.upset[,colnames(an.upset)%in%ninetymap$ID]

field_pa <- rowSums(field)>0  
control_pa <- rowSums(control) >0
thirty_pa <- rowSums(thirty) >0
sixty_pa <- rowSums(sixty) >0
ninety_pa <- rowSums(ninety) >0
an.upsetdata <- data.frame(field=field_pa, control=control_pa, thirty=thirty_pa, sixty=sixty_pa, ninety=ninety_pa)

upset(data = an.upsetdata, intersect = c("ninety", "sixty","thirty", "control", "field") ,
      name="A Horizon, nonDNase Viromes",
      min_size = 100,
      width_ratio = 0.125,
      sort_sets=FALSE,
      themes=upset_default_themes(text=element_text(size = 20)),
      set_sizes=(upset_set_size()+theme(axis.text.x = element_text(angle=90))),
      stripes=upset_stripes(colors=c('red', 'darkorange', 'gold','royalblue', 'lightblue')))# +
    # labs(title = "vOTU presence by Temperature")

ggsave("an_upset.pdf", width = 200, height = 150, units = "mm", dpi = 500)
```
```{r}
# O horizon, untreated
# upset plot for O horizon, no DNase
map <- on.map

fieldmap <- map %>%
  filter(Treatment=="Field")
controlmap <- map %>%
  filter(Treatment=="Control")
thirtymap <- map %>%
  filter(Treatment=="30ºC")
sixtymap <- map %>%
  filter(Treatment=="60ºC")
ninetymap <- map %>%
  filter(Treatment=="90ºC")

on.upset <- on.otu.rel

field <- on.upset[,colnames(on.upset)%in%fieldmap$ID]
control <- on.upset[,colnames(on.upset)%in%controlmap$ID]
thirty <- on.upset[,colnames(on.upset)%in%thirtymap$ID]
sixty <- on.upset[,colnames(on.upset)%in%sixtymap$ID]
ninety <- on.upset[,colnames(on.upset)%in%ninetymap$ID]

field_pa <- rowSums(field)>0  
control_pa <- rowSums(control) >0
thirty_pa <- rowSums(thirty) >0
sixty_pa <- rowSums(sixty) >0
ninety_pa <- rowSums(ninety) >0
on.upsetdata <- data.frame(field=field_pa, control=control_pa, thirty=thirty_pa, sixty=sixty_pa, ninety=ninety_pa)

upset(data = on.upsetdata, intersect = c("ninety", "sixty","thirty", "control", "field") ,
      name="O Horizon, nonDNase Viromes",
      min_size = 100,
      width_ratio = 0.125,
      sort_sets=FALSE,
      themes=upset_default_themes(text=element_text(size = 20)),
      set_sizes=(upset_set_size()+theme(axis.text.x = element_text(angle=90))),
      stripes=upset_stripes(colors=c('red', 'darkorange', 'gold','royalblue', 'lightblue')))# +
    # labs(title = "vOTU presence by Temperature")

ggsave("on_upset.pdf", width = 200, height = 150, units = "mm", dpi = 500)
```
```{r}
# A horizon, DNase Treated
# upset plot for A horizon, DNase
map <- ad.map

fieldmap <- map %>%
  filter(Treatment=="Field")
controlmap <- map %>%
  filter(Treatment=="Control")
thirtymap <- map %>%
  filter(Treatment=="30ºC")
sixtymap <- map %>%
  filter(Treatment=="60ºC")

ad.upset <- ad.otu.rel

field <- ad.upset[,colnames(ad.upset)%in%fieldmap$ID]
control <- ad.upset[,colnames(ad.upset)%in%controlmap$ID]
thirty <- ad.upset[,colnames(ad.upset)%in%thirtymap$ID]
sixty <- ad.upset[,colnames(ad.upset)%in%sixtymap$ID]

field_pa <- rowSums(field)>0  
control_pa <- rowSums(control) >0
thirty_pa <- rowSums(thirty) >0
sixty_pa <- rowSums(sixty) >0
ad.upsetdata <- data.frame(field=field_pa, control=control_pa, thirty=thirty_pa, sixty=sixty_pa)

upset(data = ad.upsetdata, intersect = c("sixty","thirty", "control", "field") ,
      name="A Horizon, DNase Viromes",
      min_size = 100,
      width_ratio = 0.125,
      sort_sets=FALSE,
      themes=upset_default_themes(text=element_text(size = 20)),
      set_sizes=(upset_set_size()+theme(axis.text.x = element_text(angle=90))),
      stripes=upset_stripes(colors=c('darkorange', 'gold','royalblue', 'lightblue')))# +
    # labs(title = "vOTU presence by Temperature")

ggsave("ad_upset.pdf", width = 200, height = 150, units = "mm", dpi = 500)
```
```{r}
# O horizon, DNase Treated
# upset plot for O horizon, DNase
map <- od.map

fieldmap <- map %>%
  filter(Treatment=="Field")
controlmap <- map %>%
  filter(Treatment=="Control")
thirtymap <- map %>%
  filter(Treatment=="30ºC")
sixtymap <- map %>%
  filter(Treatment=="60ºC")

od.upset <- od.otu.rel

field <- od.upset[,colnames(od.upset)%in%fieldmap$ID]
control <- od.upset[,colnames(od.upset)%in%controlmap$ID]
thirty <- od.upset[,colnames(od.upset)%in%thirtymap$ID]
sixty <- od.upset[,colnames(od.upset)%in%sixtymap$ID]

field_pa <- rowSums(field)>0  
control_pa <- rowSums(control) >0
thirty_pa <- rowSums(thirty) >0
sixty_pa <- rowSums(sixty) >0
od.upsetdata <- data.frame(field=field_pa, control=control_pa, thirty=thirty_pa, sixty=sixty_pa)

upset(data = od.upsetdata, intersect = c("sixty","thirty", "control", "field") ,
      name="O Horizon, DNase Viromes",
      min_size = 100,
      width_ratio = 0.125,
      sort_sets=FALSE,
      themes=upset_default_themes(text=element_text(size = 20)),
      set_sizes=(upset_set_size()+theme(axis.text.x = element_text(angle=90))),
      stripes=upset_stripes(colors=c('darkorange', 'gold','royalblue', 'lightblue')))# +
    # labs(title = "vOTU presence by Temperature")

ggsave("od_upset.pdf", width = 230, height = 150, units = "mm", dpi = 500)
```

# DIFFERENTIAL ABUNDANCE
# determine enrichment and cluster membership
# AD
```{r}
ad.count.75 <- daaprep(ad.otu.rel, ad.count)
ad.map.daa <- daamap(ad.map)
ad.count.75 <- daamatch(ad.map.daa,ad.count.75)
ad.map.daa <- daasetmap(ad.map.daa)
ad.count.75.1.clean <- daaclean(ad.count.75)
#check to ensure correct order
all(rownames(ad.map.daa) == colnames(ad.count.75.1.clean))
ad.dds <- tempdeseq(ad.count.75.1.clean, ad.map.daa)
ad.results <- data.frame(results(ad.dds))
ad.60.30 <- contrast_analysis(ad.dds, "60ºC", "30ºC")
ad.60.30.s <- ad.60.30 %>% filter(Sig=="S")
ad.60.C <- contrast_analysis(ad.dds, "60ºC", "Control")
ad.60.C.s <- ad.60.C %>% filter(Sig=="S")
ad.60.F <- contrast_analysis(ad.dds, "60ºC", "Field")
ad.60.F.s <- ad.60.F %>% filter(Sig=="S")
ad.30.C <- contrast_analysis(ad.dds, "30ºC", "Control")
ad.30.C.s <- ad.30.C %>% filter(Sig=="S")
ad.30.F <- contrast_analysis(ad.dds, "30ºC", "Field")
ad.30.F.s <- ad.30.F %>% filter(Sig=="S")
ad.C.F <- contrast_analysis(ad.dds, "Control", "Field")
ad.C.F.s <- ad.C.F %>% filter(Sig=="S")
ad.dds.all <- rbind(ad.C.F, ad.30.C, ad.30.F, ad.60.C, ad.60.30, ad.60.F)
ad.dds.bon <- padjbon(ad.dds.all)
ad.dds.bon.s <- sigbon(ad.dds.bon)
#get the number of unique differentially abundant vOTUs
ad.dds.bon.s$OTU %>% unique() %>% length() #2755
# create dataframe that just has unique vOTUs that are significant
ad.dds.uniq <- ad.dds.bon.s %>% distinct(OTU, .keep_all = FALSE)
ad.otu.avg <- avgreps(ad.otu, ad.map)
ad.otu.avg.s <- avgsig(ad.otu.avg, ad.dds.uniq)
ad.data.scale2 <- datascale(ad.otu.avg.s)
# calculate Z-score
ad.data.plots <- ad.data.scale2 %>%
  rownames_to_column(var='OTU') %>%
  pivot_longer(-OTU, names_to='Treatment', values_to = 'Zscore')
kmeansclust(ad.data.scale2)
fviz_nbclust(ad.data.scale2, kmeans, method = "silhouette", k.max = 24) + theme_minimal() + ggtitle("The Silhouette Plot")
# determine which number k cluster to choose
k <- 2
ad.plotz <- clusterz(ad.data.scale2, ad.data.plots)
ad.clusterplotz <- plotclusters(ad.plotz, title = "A horizon, DNase", filename = "ad_clusterplotz.pdf")
ad.plotz.uniq <- clustmember(ad.plotz, subgroup = "AD")
ad.clupset <- clustupset(ad.plotz.uniq)
```
# OD
```{r}
od.count.75 <- daaprep(od.otu.rel, od.count)
od.map.daa <- daamap(od.map)
od.count.75 <- daamatch(od.map.daa,od.count.75)
od.map.daa <- daasetmap(od.map.daa)
od.count.75.1.clean <- daaclean(od.count.75)
#check to ensure correct order
all(rownames(od.map.daa) == colnames(od.count.75.1.clean))
od.dds <- tempdeseq(od.count.75.1.clean, od.map.daa)
od.results <- data.frame(results(od.dds))
od.60.30 <- contrast_analysis(od.dds, "60ºC", "30ºC")
od.60.30.s <- od.60.30 %>% filter(Sig=="S")
od.60.C <- contrast_analysis(od.dds, "60ºC", "Control")
od.60.C.s <- od.60.C %>% filter(Sig=="S")
od.60.F <- contrast_analysis(od.dds, "60ºC", "Field")
od.60.F.s <- od.60.F %>% filter(Sig=="S")
od.30.C <- contrast_analysis(od.dds, "30ºC", "Control")
od.30.C.s <- od.30.C %>% filter(Sig=="S")
od.30.F <- contrast_analysis(od.dds, "30ºC", "Field")
od.30.F.s <- od.30.F %>% filter(Sig=="S")
od.C.F <- contrast_analysis(od.dds, "Control", "Field")
od.C.F.s <- od.C.F %>% filter(Sig=="S")
od.dds.all <- rbind(od.C.F, od.30.C, od.30.F, od.60.C, od.60.30, od.60.F)
od.dds.bon <- padjbon(od.dds.all)
od.dds.bon.s <- sigbon(od.dds.bon)
#get the number of unique differentially abundant vOTUs
od.dds.bon.s$OTU %>% unique() %>% length() #3219
# create dataframe that just has unique vOTUs that are significant
od.dds.uniq <- od.dds.bon.s %>% distinct(OTU, .keep_all = FALSE)
od.otu.avg <- avgreps(od.otu, od.map)
od.otu.avg.s <- avgsig(od.otu.avg, od.dds.uniq)
od.data.scale2 <- datascale(od.otu.avg.s)
# calculate Z-score
od.data.plots <- od.data.scale2 %>%
  rownames_to_column(var='OTU') %>%
  pivot_longer(-OTU, names_to='Treatment', values_to = 'Zscore')
kmeansclust(od.data.scale2)
fviz_nbclust(od.data.scale2, kmeans, method = "silhouette", k.max = 24) + theme_minimal() + ggtitle("The Silhouette Plot")
# determine which number k cluster to choose
k <- 2
od.plotz <- clusterz(od.data.scale2, od.data.plots)
od.clusterplotz <- plotclusters(od.plotz, title = "O horizon, DNase", filename = "od_clusterplotz.pdf")
od.plotz.uniq <- clustmember(od.plotz, subgroup = "OD")
od.clupset <- clustupset(od.plotz.uniq)
```
# DNASE - investigating membership in trend groups
```{r}
d.clupset <- merge(ad.clupset, od.clupset, all = TRUE)
d.clupset[is.na(d.clupset)] <- FALSE
ad.od.dds.all <- clustpresent2(ad.dds.all, od.dds.all)
# Check all cluster vOTUs are present in all subgroups
all_vOTUs_present <- all(d.clupset$OTU %in% ad.od.dds.all$OTU)
print(all_vOTUs_present)
# if FALSE, identify vOTUs that aren't present and then remove
d.all.clupset <- d.clupset %>%
  inner_join(ad.od.dds.all, by = "OTU")
d.all.clupset <- d.all.clupset %>%
  dplyr::rename(
    "Tolerant_A" = "2_AD",
    "Tolerant_O" = "2_OD",
    "Sensitive_A" = "1_AD",
    "Sensitive_O" = "1_OD")
upset(data = d.all.clupset, intersect = c("Tolerant_A", "Tolerant_O", "Sensitive_A", "Sensitive_O") ,
      name = "Heat Tolerant or Heat Sensitive Groups", 
      min_size = 0, width_ratio = 0.125, sort_sets=FALSE,
      set_sizes=(upset_set_size()+theme(axis.text.x = element_text(angle=90))),
      themes=upset_default_themes(text=element_text(size = 20)),
      stripes=upset_stripes(colors=cluster_colors_upset_x2)) #+
    #labs(title = )
ggsave("d.all.clupset.pdf", width = 220, height = 140, units = "mm", dpi = 500)
```

```{r}
# vcontact prepping of indicator vOTUs data frame
od.stats <- od.dds.bon %>%
  dplyr::rename(
    "Genome" = "gene")
od.stats <- od.stats %>%
  select("Genome", "statistic", "p.adj2")

od.plotz.statz <- od.plotz.uniq %>%
  dplyr::rename(
    "Genome" = "OTU")

o_indicators <- merge(od.plotz.statz, od.stats, by = "Genome", all = TRUE)

o_indicators <- o_indicators %>%
  mutate(Cluster_subgroup = ifelse(is.na(Cluster_subgroup), "NS", Cluster_subgroup))
o_indicators <- o_indicators %>%
  mutate(O_Response = case_when(
    Cluster_subgroup == "1_OD" ~ "O_Sensitive",
    Cluster_subgroup == "2_OD" ~ "O_Tolerant", 
    Cluster_subgroup == "NS" ~ "NS"))
o_indicators <- o_indicators %>%
  select(-Cluster_subgroup)

saveRDS(o_indicators, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/o_indicators.RDS")



ad.stats <- ad.dds.bon %>%
  dplyr::rename(
    "Genome" = "gene")
ad.stats <- ad.stats %>%
  select("Genome", "statistic", "p.adj2")

ad.plotz.statz <- ad.plotz.uniq %>%
  dplyr::rename(
    "Genome" = "OTU")

a_indicators <- merge(ad.plotz.statz, ad.stats, by = "Genome", all = TRUE)
a_indicators <- a_indicators %>%
  mutate(Cluster_subgroup = ifelse(is.na(Cluster_subgroup), "NS", Cluster_subgroup))
a_indicators <- a_indicators %>%
  mutate(A_Response = case_when(
    Cluster_subgroup == "1_AD" ~ "A_Sensitive",
    Cluster_subgroup == "2_AD" ~ "A_Tolerant", 
    Cluster_subgroup == "NS" ~ "NS"))
a_indicators <- a_indicators %>%
  select(-Cluster_subgroup)

a_o_indicators <- merge(a_indicators, o_indicators, by = "Genome", all = TRUE)
a_o_indicators <- a_o_indicators[complete.cases(a_o_indicators), ]
a_o_indicators <- a_o_indicators %>%
  mutate(Response = case_when(
    O_Response == "O_Sensitive" & A_Response == "A_Sensitive" ~ "Sensitive",
    O_Response == "O_Tolerant" & A_Response == "A_Tolerant" ~ "Tolerant",
    O_Response == "O_Sensitive" & A_Response == "A_Tolerant" ~ "Mixed",
    O_Response == "O_Tolerant" & A_Response == "A_Sensitive" ~ "Mixed",
    O_Response == "O_Tolerant" & A_Response == "NS" ~ "O_Tolerant",
    O_Response == "O_Sensitive" & A_Response == "NS" ~ "O_Sensitive",
    O_Response == "NS" & A_Response == "A_Tolerant" ~ "A_Tolerant",
    O_Response == "NS" & A_Response == "A_Sensitive" ~ "A_Sensitive",
    O_Response == "NS" & A_Response == "NS" ~ "NS"))

saveRDS(a_o_indicators, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/a_o_indicators.RDS")

```



# HOST PREDICTION
```{r}
iphop.genus <- read.csv("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/all_Host_prediction_to_genus_m90.csv")
iphop.filter <- iphop.genus %>%
  group_by(Virus) %>%
  filter(Confidence.score == max(Confidence.score)) %>%
  ungroup()
colnames(iphop.filter)[colnames(iphop.filter) == "Virus"] <- "OTU"

# to determine if there are duplicates
length(unique(iphop.filter$OTU))
# remove exact duplicates
iphop.uniq <- iphop.filter %>%
  distinct()
# to investigate the other duplicates use this and keep repeating until all duplicates are taken care of
duplicates <- iphop.uniq %>% filter(duplicated(OTU) | duplicated(OTU, fromLast = TRUE))
# remove duplicates that are the same except for different HostGenus and change that value to "mixed"
iphop.uniq <- iphop.uniq %>%
  group_by(OTU) %>%
  mutate(HostGenus = ifelse(n_distinct(HostGenus) > 1, "mixed", HostGenus)) %>%
  ungroup()
# remove duplicates that are the same except for List.of.methods and just take the one that is better (need to specify)
iphop.uniq <- iphop.uniq %>%
  filter(!(OTU == "C9_S57_L004_contig_2815" & List.of.methods == "iPHoP-RF;97.10 CRISPR;88.50"))
iphop.uniq <- iphop.uniq %>%
  filter(!(OTU == "HD2_S9_L004_contig_918" & List.of.methods == "iPHoP-RF;96.40"))
iphop.uniq <- iphop.uniq %>%
  filter(!(OTU == "HD5_S15_L004_contig_1958" & List.of.methods == "iPHoP-RF;97.10"))
# there is still one duplicate left, but no clear reason to pick one or the other...so I'm randomly just picking one of them
iphop.uniq <- iphop.uniq %>%
  filter(!(OTU == "C9_S57_L004_contig_4327" & HostPhylum == "Cyanobacteria"))
# remove exact duplicates
iphop.uniq <- iphop.uniq %>%
  distinct()

#create dataframe with just OTU and phylum from iphop
iphop.phylum <- iphop.uniq[, c("OTU", "HostPhylum")]
# name row.names of otu.rel
otu.rel.otu <- otu.rel %>%
  rownames_to_column(var = "OTU")

otu.rel.phylum <- merge(iphop.phylum, otu.rel.otu, by = "OTU", all.y = TRUE)

# replace NA Host Phylum with "unknown"
otu.rel.phylum <- otu.rel.phylum %>% mutate(HostPhylum = ifelse(is.na(HostPhylum), "Unknown", HostPhylum))
# remove OTU column (maybe not)
otu.rel.phylum <- otu.rel.phylum %>% select(-OTU) 

otu.phylum.long <- pivot_longer(otu.rel.phylum, cols=2:55, names_to = "ID", values_to = "RelAbd")

otu.phylum.stack <- group_by(otu.phylum.long, HostPhylum, ID) %>%
  summarize(RelAbd=sum(RelAbd))
# Prep data frame for plotting
otu.phylum.map <- merge(otu.phylum.stack, metadata, by="ID")

# Plot
hostphylum <- ggplot(otu.phylum.map, aes(x=ID, y=RelAbd, fill=HostPhylum))+
    labs(x="Sample ID", 
       y ="Relative Abundance",
       title = "Heating vOTU Host Prediction by Sample")+
  geom_bar(stat="identity", position = "fill")+
  scale_fill_manual(values=phylum_palette) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
hostphylum <- ggsave("allhostheat_phylum.png", width = 8.6, height = 6, units = "in", dpi = 500)
```

```{r}
# O horizon, DNase
# name row.names of otu.rel
od.otu.rel.otu <- od.otu.rel %>%
  rownames_to_column(var = "OTU")

od.otu.rel.phylum <- merge(iphop.phylum, od.otu.rel.otu, by = "OTU", all.y = TRUE)
# replace NA Host Phylum with "unknown"
od.otu.rel.phylum <- od.otu.rel.phylum %>% mutate(HostPhylum = ifelse(is.na(HostPhylum), "Unknown", HostPhylum))
# remove OTU column (maybe not)
od.otu.rel.phylum <- od.otu.rel.phylum %>% select(-OTU) 

od.otu.phylum.long <- pivot_longer(od.otu.rel.phylum, cols=2:13, names_to = "ID", values_to = "RelAbd")

od.otu.phylum.stack <- group_by(od.otu.phylum.long, HostPhylum, ID) %>%
  summarize(RelAbd=sum(RelAbd))

od.otu.phylum.stack <- od.otu.phylum.stack %>%
  group_by(HostPhylum) %>%
  mutate(NewHostPhylum = ifelse(all(RelAbd < 0.001), 'Other', HostPhylum)) %>%
  ungroup()

# Prep data frame for plotting
od.otu.phylum.map <- merge(od.otu.phylum.stack, od.map, by="ID")

od.otu.phylum.map <- od.otu.phylum.map %>% 
  mutate(Order = case_when(
    Treatment == "Field" ~ "1",
    Treatment == "Control" ~ "2", 
    Treatment == "30ºC" ~ "3", 
    Treatment == "60ºC" ~ "4")) %>%
  arrange(Order) %>%
  mutate(TreatRep = paste(Treatment, Replicate))

od.otu.phylum.map$TreatRep <- factor(od.otu.phylum.map$TreatRep, levels = unique(od.otu.phylum.map$TreatRep))

unique(od.otu.phylum.map$HostPhylum)

# Plot
temp2 <- od.otu.phylum.map[od.otu.phylum.map$HostPhylum != "Unknown", ]

ODhostphylum2 <- ggplot(temp2, aes(x=TreatRep, y=RelAbd, fill=factor(NewHostPhylum, level=phyla_order)))+
    labs(x="Sample", 
       y ="Relative Abundance",
       title = "vOTU Host Prediction by Sample (O, DNase)")+
  geom_bar(stat="identity")+
  scale_fill_manual(values=collapsed_phylum_colors, limits = phyla_order) +
  geom_vline(xintercept = c(3.5, 6.5, 9.5), linetype = "solid", color = "black", size = 1)+
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1), 
        legend.position = "none")
ODhostphylum2
ODhostphylum2 <- ggsave("ODhostheat2_phylum.pdf", width = 200, height = 150, units = "mm", dpi = 500)
```


```{r}
# A horizon, DNase
# name row.names of otu.rel
ad.otu.rel.otu <- ad.otu.rel %>%
  rownames_to_column(var = "OTU")

ad.otu.rel.phylum <- merge(iphop.phylum, ad.otu.rel.otu, by = "OTU", all.y = TRUE)
# replace NA Host Phylum with "unknown"
ad.otu.rel.phylum <- ad.otu.rel.phylum %>% mutate(HostPhylum = ifelse(is.na(HostPhylum), "Unknown", HostPhylum))
# remove OTU column (maybe not)
ad.otu.rel.phylum <- ad.otu.rel.phylum %>% select(-OTU) 

ad.otu.phylum.long <- pivot_longer(ad.otu.rel.phylum, cols=2:13, names_to = "ID", values_to = "RelAbd")

ad.otu.phylum.stack <- group_by(ad.otu.phylum.long, HostPhylum, ID) %>%
  summarize(RelAbd=sum(RelAbd))

ad.otu.phylum.stack <- ad.otu.phylum.stack %>%
  group_by(HostPhylum) %>%
  mutate(NewHostPhylum = ifelse(all(RelAbd < 0.001), 'Other', HostPhylum)) %>%
  ungroup()
# Prep data frame for plotting
ad.otu.phylum.map <- merge(ad.otu.phylum.stack, ad.map, by="ID")

ad.otu.phylum.map <- ad.otu.phylum.map %>% 
  mutate(Order = case_when(
    Treatment == "Field" ~ "1",
    Treatment == "Control" ~ "2", 
    Treatment == "30ºC" ~ "3", 
    Treatment == "60ºC" ~ "4")) %>%
  arrange(Order) %>%
  mutate(TreatRep = paste(Treatment, Replicate))

ad.otu.phylum.map$TreatRep <- factor(ad.otu.phylum.map$TreatRep, levels = unique(ad.otu.phylum.map$TreatRep))

unique(ad.otu.phylum.map$HostPhylum)

# Plot
temp2 <- ad.otu.phylum.map[ad.otu.phylum.map$HostPhylum != "Unknown", ]

ADhostphylum2 <- ggplot(temp2, aes(x=TreatRep, y=RelAbd, fill=factor(NewHostPhylum, level=phyla_order)))+
    labs(x="Sample", 
       y ="Relative Abundance",
       title = "vOTU Host Prediction by Sample (A, DNase)")+
  geom_bar(stat="identity")+
  scale_fill_manual(values=collapsed_phylum_colors, limits = phyla_order) +
  geom_vline(xintercept = c(3.5, 6.5, 9.5), linetype = "solid", color = "black", size = 1)+
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1), 
        legend.position = "none")
ADhostphylum2
ADhostphylum2 <- ggsave("ADhostheat2_phylum.pdf", width = 200, height = 150, units = "mm", dpi = 500)
```

# IPHOP STATS
```{r}
od.host.stats <- od.otu.phylum.map

actino_od <- od.host.stats %>%
  filter(HostPhylum == "Actinobacteriota")
acido_od <- od.host.stats %>%
  filter(HostPhylum == "Acidobacteriota")
bact_od <- od.host.stats %>%
  filter(HostPhylum == "Bacteroidota")
cyan_od <- od.host.stats %>%
  filter(HostPhylum == "Cyanobacteria")
proteo_od <- od.host.stats %>%
  filter(HostPhylum == "Proteobacteria")

kruskal.test(RelAbd ~ Treatment, actino_od)
kruskal.test(RelAbd ~ Treatment, acido_od)
kruskal.test(RelAbd ~ Treatment, bact_od)
kruskal.test(RelAbd ~ Treatment, cyan_od)
kruskal.test(RelAbd ~ Treatment, proteo_od)


ad.host.stats <- ad.otu.phylum.map

actino_ad <- ad.host.stats %>%
  filter(HostPhylum == "Actinobacteriota")
acido_ad <- ad.host.stats %>%
  filter(HostPhylum == "Acidobacteriota")
bact_ad <- ad.host.stats %>%
  filter(HostPhylum == "Bacteroidota")
proteo_ad <- ad.host.stats %>%
  filter(HostPhylum == "Proteobacteria")

kruskal.test(RelAbd ~ Treatment, actino_ad)
kruskal.test(RelAbd ~ Treatment, acido_ad)
kruskal.test(RelAbd ~ Treatment, bact_ad)
kruskal.test(RelAbd ~ Treatment, proteo_ad)
```

# Create IPHOP Host Phyla Legend
```{r}
collapsed_host_phyla_colors <- data.frame(
  host_phylum = c("Acidobacteriota", "Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Proteobacteria", "Other"),
  colors = c("#7526C3","#75d644","#FAE500","#1CFEDA","#D78EFC","darkgray")
)

host_phyla_order <- c("Acidobacteriota", "Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Proteobacteria", "Other")

collapsed_host_phyla_legend <- ggplot(collapsed_host_phyla_colors, aes(x = 1, y = factor(host_phylum, level=host_phyla_order), fill = host_phylum)) +
  geom_tile(show.legend = TRUE) +
  scale_fill_manual(values = collapsed_host_phyla_colors$colors, limits = host_phyla_order) +
  labs(fill = "Host Phylum")+
  theme(legend.position = "bottom")
collapsed_host_phyla_legend + guides(fill = guide_legend(title.position = "top", ncol = 6))

collapsed_host_phyla_legend <- ggsave("collapsed_host_phylalegend.pdf", width = 250, height = 200, units = "mm", dpi = 500)
```



