---
title: "vContact"
output: html_document
date: "2023-11-27"
---
Load libraries
```{r}
library(igraph)
library(GGally)
library(tidyverse)
```

Load data 
We need two files:
- Network file - this is the c1.ntw file generated by vConTACT2 and contains each pair of viral genomes with a significant overlap in their gene content (shared proteins).
- Functional annotation file - a data frame containing a list of vOTUs (first column) and associated attributes. 
```{r}
#some statistical method to deal with "problems" associated with multiple comparisons
adj.method <- "holm"

ntwk <-read.table("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/c1.ntw", header = F, sep = " ", col.names = c("OTU1", "OTU2", "Score"))

a_o_indicators <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/a_o_indicators.RDS")
fnct.trt <- a_o_indicators

fnct.trt %>% 
  group_by(Response) %>% 
  dplyr::count()
```

Filter data
Make sure to remove all vOTUs that are not shared across the nodes, edges, and functional annotation data frames
```{r}
fnct.trt.filt <- fnct.trt %>% 
  filter(Genome %in% ntwk$OTU1 | Genome %in% ntwk$OTU2)

good.genomes <- unique(fnct.trt.filt$Genome)
good.genomes

ntwk.filt <- ntwk  %>%
  filter(OTU1 %in% good.genomes & OTU2 %in% good.genomes)
```

Generate the 2D network layout with the filtered data
Note: Make sure to use a force-directed network layout algorithm to generate the 2D configuration of the network. 
The two modes available in ggnet2 are the "Fruchterman-Reingold" (mode = "fruchtermanreingold") and the Kamada-Kawai (mode = "kamadakawai") algorithms. 
```{r}
layout <- "fruchtermanreingold"

nodes <- GGally::ggnet2(as.matrix(ntwk.filt[,-3]), 
                mode = layout, 
                layout.par = list(list=(niter=2000))) %>% 
  .$data %>% 
  dplyr::rename("Genome" = "label")

edges <- ntwk.filt %>% 
  mutate(Pair = paste(OTU1, OTU2, sep = ".")) %>% 
  gather(key = "Member", value = "Genome", -Pair, -Score) %>% 
  inner_join(nodes, by = "Genome")
```

Generate igraph object
Note: When calculating weight-based distances, igraph sums the weights linked to each pair. Since the weights generated by vConTACT2 are the -log10(P-vals) of hypergeometric tests, higher values denote closer similarities between pairs of genomes. Therefore, we need to transform the vConTACT scores into inverse values (weight = 1/Score) so that the shortest paths found by igraph are between more closely related nodes. An alternative is to transform the vConTACT scores to P-values but I have found that they can be extremely small (10^-300) which can mess up with the neighborhood calculations (igraph turns very small values into zeros). Potentially, we could multiply the P-values by the inverse of the smallest P-value (weight = (10^-Score)*10^max(ntwk.filt$Score)) but I haven't checked if that messes up with downstream calculations
```{r}
edges.igraph <- ntwk.filt %>% 
  mutate(weight = 1/Score) %>% 
  select(OTU1, OTU2, weight) 

nodes.igraph <- select(nodes, Genome) 

ntwk.igraph <- graph_from_data_frame(d = edges.igraph, vertices = nodes.igraph, directed = F)
```

Calculate distances
```{r}
### Weighted distances
ntwk.dist <- distances(ntwk.igraph) 

#inf means there is an unreachable or infinite distance between nodes (no direct path/edge between the two nodes)
```

Calculate distance threshold
Plot distribution of distances
```{r}
dist.filt <- ntwk.dist[upper.tri(ntwk.dist)]  ## We want to get rid of redundant values by removing the diagonal and the lower triangle before calculating the minimum distance

min.dist <- quantile(dist.filt, .01)
good.dist <- ntwk.dist < min.dist

edges %>% 
  mutate(weight = 1/Score) %>% 
  ggplot(aes(weight)) +
  geom_histogram(fill = "gray85") + 
  theme_bw()

edges %>% 
  ggplot(aes(Score)) +
  geom_histogram(fill = "gray85") +
  theme_bw()

ggplot(data = data.frame(Distance = dist.filt), aes(Distance)) + 
  geom_histogram(fill = "gray85") + 
  geom_vline(xintercept = c(min.dist), linetype = 2) +
  theme_bw()
```

Identify the neighborhood based on the distance threshold
```{r}
nodes.ids <- nodes$Genome %>% unique()

#initialize empty lists
neighborhood.list <- list()
nhood.length.list <- list()

# populate lists
for(node in nodes.ids){
  neighborhood <- good.dist[,node][good.dist[,node] == T] %>% names()
  neighborhood.list[[node]] <- neighborhood
  nhood.length.list[[node]] <- length(neighborhood)
}
# 1,995 neighborhoods

nhood.length <- plyr::ldply(nhood.length.list, function(x) x)
names(nhood.length) <- c("Neighborhood", "Size")

nhood.length %>% 
  ggplot(aes(Size)) +
  geom_histogram(fill = "gray85", binwidth = 5) + 
  geom_vline(xintercept = 5, linetype = 2) +
  theme_bw()

ggsave("nhooddistribution.png", width = 10, height = 5, units = "in", dpi = 500)
```

Calculate enriched neighborhoods for each trait 
```{r}
sig.otus <- fnct.trt.filt %>%
  filter(Response != "NS") %>%
  mutate(Trait = Response)

enrichment.list.trt <- list()

for(trait in unique(sig.otus$Trait)){
  sig.trait <- filter(sig.otus, Trait == trait)$Genome
  hgt.list <- list()
  for(node in nodes.ids){
    neighborhood <- neighborhood.list[[node]]
    neighborhood.positive <- sum(neighborhood %in% sig.trait)
    universe.positive <- sum(nodes.ids %in% sig.trait)
    universe.negative <- sum(!nodes.ids %in% sig.trait)
    neighborhood.size<- length(neighborhood)
    hgt.list[[node]] <- phyper(neighborhood.positive, universe.positive, universe.negative, neighborhood.size, lower.tail = F)
  }
  hgt <- plyr::ldply(hgt.list, function(x) x)
  names(hgt) <- c("Neighborhood", "pval")
  enrichment.list.trt[[trait]] <- hgt
}

enrichment.trt <- plyr::ldply(enrichment.list.trt, function(x) x) %>% 
  dplyr::rename("Trait" = ".id") %>% 
  ungroup() %>% 
  inner_join(nhood.length, "Neighborhood") %>% 
  filter(Size >= 10) %>% 
  mutate(padj = p.adjust(pval, method = adj.method)) %>% 
  group_by(Trait, Neighborhood) %>% 
  mutate(logpval = -log10(padj),
         EnrichmentScore = min(-log10(padj), -log10(10^-5))/-log10(10^-5)) %>% 
  ungroup() %>% 
  arrange(-EnrichmentScore) %>% mutate(EnrichmentBin = case_when(EnrichmentScore < 0.2 ~ 1,
                                                EnrichmentScore < 0.4 ~ 2,
                                                EnrichmentScore < 0.6 ~ 3,
                                                EnrichmentScore < 0.8 ~ 4,
                                                TRUE ~ 5)) 

#Inspect results
enrichment.trt %>% 
  filter(padj < 0.05) %>% 
  group_by(Trait) %>% 
  dplyr::count()
```

Generate a data frame with all the members for each neighborhood
```{r}
neighborhood.tidy.list <- list()

for(node in nodes.ids){
  neighborhood.tidy.list[[node]] <- data.frame(Members = neighborhood.list[[node]])
}

neighborhood.tidy <- plyr::ldply(neighborhood.tidy.list, function(x) x) %>% 
  dplyr::rename("Neighborhood" = ".id") 
```

Identify the subnetwork that aggregates all the significant neighborhoods with an enrichment of heat enriched vOTUs and generate a new network layout for plotting
```{r}
# wait... so the above network is everything and then what I am about to create is just significant neighborhoods for only heat tolerant?? yes, I believe so...

traits <- c("O_Tolerant", "Tolerant", "A_Tolerant", "Mixed")

int.neighborhoods <- filter(enrichment.trt, Trait %in% traits) %>% 
  filter(padj < 0.05)

int.subntwk <- neighborhood.tidy %>% 
  filter(Neighborhood %in% int.neighborhoods$Neighborhood) %>% 
  group_by(Members) %>% 
  dplyr::count() %>% 
  .$Members

nbhd.nodes <- GGally::ggnet2(as.matrix(filter(ntwk.filt, OTU1 %in% int.subntwk & OTU2 %in% int.subntwk)[,-3]), 
                mode = layout, 
                layout.par = list(list=(niter=2000))) %>% 
  .$data %>% 
  dplyr::rename("Genome" = "label")

nbhd.edges <- ntwk.filt %>% 
  mutate(Pair = paste(OTU1, OTU2, sep = ".")) %>% 
  gather(key = "Member", value = "Genome", -Pair, -Score) %>% 
  inner_join(nbhd.nodes, by = "Genome")
```

Save results
```{r}
saveRDS(nodes, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/safe_ind_whole_nodes.RDS")
saveRDS(edges, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/safe_ind_whole_edges.RDS")
saveRDS(nbhd.nodes, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/safe_ind_subntwk_nodes.RDS")
saveRDS(nbhd.edges, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/safe_ind_subntwk_edges.RDS")
saveRDS(enrichment.trt, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/safe_ind_enrichment.RDS")
saveRDS(neighborhood.tidy, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/safe_ind_eighborhoods.RDS")
```

# load OTU
```{r}
# LOAD ABUNDANCE DATA FOR ENTIRE DATASET
otu=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/all.75.tmean.tsv", delim = "\t", col_names = TRUE)

# PREP DATA FRAME
otu <- col_to_row(otu, "Contig", "contignames")
# edit column names 
colnames(otu) <- gsub(pattern=".vib.sI.Trimmed.Mean", replacement = "", x=colnames(otu))
# remove two samples that are not necessary
otu$N1_S61_L004 <- NULL
otu$N2_S62_L004 <- NULL

otu.tidy <- otu %>%
  rownames_to_column(var = "OTU_ID") %>%
  pivot_longer(cols=-c(OTU_ID), names_to = "SampleID", values_to = "Abundance")

```


# Genomes and Clusters
```{r}
# upload the genome_by_genome_overview file generated by vContact2, which contains information on virus clusters and individual genomes and their membership in clusters and their taxonomy
genome.ov <- read.table("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/genome_by_genome_overview.csv", header = T, sep = ",")

#Get the source of each contig
genome.ov <- genome.ov %>% 
  mutate(Source = ifelse(str_detect(Genome, "contig"), "heat", "refseq"))

# Check if all the contigs are accounted for. This was chris's issue: Total should be 2962 but there's only 2948 in the vcontact output. In order have the right numbers, To make sure the final OTU master file includes those missing genomes, I added them and classify the as singletons.
# Dont need to do this because the number is the same

genome.ov %>% 
  filter(Source == "heat") %>% 
  group_by(Genome) %>% 
  dplyr::count()

missing.ids <- filter(otu.tidy, !OTU_ID %in% genome.ov$Genome) %>%
  .$OTU_ID %>%
  unique()

# nothing is missing

#missing.df <- data.frame(Genome = missing.ids,
                         #VC.Status = "Singleton",
                         #Source = "heat")
#missing.df

# Also, some the OTUs are duplicated and either classified as clustered or either singletons / outliers.  In each case, the clustered record doesn't have a VC assigned. Not sure what's going on here but I'll just classify them as singletons or outliers for downstream analyses. 

genome.ov %>% 
  group_by(Genome) %>% 
  mutate(Duplicates = n()) %>% 
  filter(Duplicates > 1) %>% 
  arrange(Genome)

duplicated.recs <- genome.ov %>% 
  group_by(Genome) %>% 
  mutate(Duplicates = n()) %>% 
  filter(Duplicates > 1) %>% 
  filter(VC.Status != "Clustered") %>% 
  select(-Duplicates) 

good.recs <- genome.ov %>% 
  filter(!Genome %in% duplicated.recs$Genome)

genome.master <- bind_rows(good.recs, duplicated.recs)
```

Generate a data frame that specifies the composition of each cluster in terms of source of network nodes
```{r}
clstr.source <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  mutate(heat = str_detect(Genome, "contig")) %>% 
  group_by(VC, Size) %>% 
  summarise(pheat = sum(heat)/n()) %>% 
  mutate(ClstrComp = case_when(pheat == 0 ~ "refseq",
                               pheat == 1 ~ "heat",
                               TRUE ~ "both" )) 
```
Let's create a data frame with the consensus viral taxonomy for each cluster. The strategy is to check each rank and see if there's only one assignment or a mix. Some refeseq entrees have unassigned ranks (see VC 104_0) so I'm ignoring those and deciding the proper classification based on the rest of the VC members 
```{r}
clstr.order <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Order != "Unassigned") %>% 
  group_by(VC, Order) %>% 
  dplyr::count() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Order = ifelse(Duplicates > 1, "Mixed", as.character(Order))) %>% 
  group_by(VC, Order) %>% 
  dplyr::count() %>% 
  select(-n)

clstr.family <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Family != "Unassigned") %>% 
  group_by(VC, Family) %>% 
  dplyr::count() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Family = ifelse(Duplicates > 1, "Mixed", as.character(Family))) %>% 
  group_by(VC, Family) %>% 
  dplyr::count() %>% 
  select(-n)

clstr.genus <-  genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Genus != "Unassigned") %>% 
  group_by(VC, Genus) %>% 
  dplyr::count() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Genus = ifelse(Duplicates > 1, "Mixed", as.character(Genus))) %>% 
  group_by(VC, Genus) %>% 
  dplyr::count() %>% 
  select(-n)
```


Let's get the taxonomy of the host associated to refseq by extracting the genus from the ID and getting the higher taxa from greengenes taxonomy file.
FYI - The greengenes taxonomy has some weird assignments that need to be carefully inspected. For example, Enterococcus is both classified as Actinobacteria and Firmicutes. 
I selected the most prevalent taxonomy for those hosts with conflicting higher taxonomic ranks. 
```{r}
host.tax <- read.table("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/gg_13_8_97_tax.tsv", header = T, sep = "\t", quote = "")

genus.host <- genome.ov %>% 
  filter(Source == "refseq") %>% 
  group_by(Genome) %>% 
  dplyr::count() %>% 
  select(-n) %>% 
  separate(Genome, c("Genus"),sep = "~", fill = "right", remove = F) 

refseq.tmp <- host.tax %>% 
  filter(Genus %in% genus.host$Genus) %>% 
  group_by(Phylum, Class, Order, Family, Genus) %>% 
  dplyr::count() %>% 
  arrange(Genus) %>% 
  group_by(Genus) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Keep = ifelse(Duplicates == 1, TRUE, ifelse(n == max(n), TRUE, FALSE))) %>% 
  ungroup() %>% 
  filter(Keep) %>% 
  select(-n, -Duplicates, -Keep) %>% 
  right_join(genus.host, by = "Genus") %>% 
  mutate(HostPhylum = ifelse(is.na(Phylum), "Unassigned", as.character(Phylum)),
         HostClass = ifelse(is.na(Class), "Unassigned", as.character(Class)),
         HostOrder= ifelse(is.na(Order), "Unassigned", as.character(Order)),
         HostFamily = ifelse(is.na(Family), "Unassigned", as.character(Family)),
         HostGenus = as.character(Genus)) %>% 
  select(Genome, HostPhylum:HostGenus)
  
refseq.good <- filter(refseq.tmp, HostPhylum != "Unassigned")
refseq.bad <- filter(refseq.tmp, HostPhylum == "Unassigned")
```

Fixing the taxonomies. I am assuming enterobacteria are all enterobacteriaceae
```{r}
missing.tax <- tribble(
  ~HostPhylum, ~HostClass, ~HostOrder, ~HostFamily, ~HostGenus,
  "Proteobacteria", "Betaproteobacteria", "Burkholderiales", "Alcaligenaceae", "Bordetella",
  "Proteobacteria", "Alphaproteobacteria", "Rhizobiales", "Brucellaceae", "Brucella",
  "Bacteroidetes", "Flavobacteriia", "Flavobacteriales", "Flavobacteriaceae", "Croceibacter",
  "Cyanobacteria", "Unassigned", "Unassigned", "Unassigned", "Cyanophage",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Enterobacteria",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Enterobacterial",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Enterobacteriaphage",
  "Proteobacteria", "Gammaproteobacteria", "Alteromonadales", "Idiomarinaceae", "Idiomarinaceae",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Kluyvera",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Lelliottia",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Pectobacteriaceae", "Pectobacterium",
  "Proteobacteria", "Alphaproteobacteria", "Rickettsiales", "Pelagibacteriaceae", "Pelagibacter",
  "Proteobacteria", "Gammaproteobacteria", "Oceanospirillales", "	Halomonadaceae", "Salicola",
  "Proteobacteria", "Alphaproteobacteria", "Rhodobacterales", "Rhodobacteraceae", "Silicibacter",
  "Tenericutes", "Mollicutes", "Entomoplasmatales", "Spiroplasmataceae", "Spiroplasma",
  "Crenarchaeota", "Thermoprotei", "Sulfolobales", "Unassigned", "Sulfolobales",
  "Crenarchaeota", "Thermoprotei", "Thermoproteales", "Thermoproteaceae", "Thermoproteus",
  "Verrucomicrobia", "Unassigned", "Unassigned", "Unassigned", "Verrucomicrobia"
)

refseq.manual <- refseq.bad %>% 
  select(Genome, HostGenus) %>% 
  inner_join(missing.tax, by = "HostGenus") %>% 
  select(-HostGenus, HostGenus)

refseq.manual

```

Correcting typos
```{r}
refseq.bad <- refseq.bad %>% 
  select(Genome, HostGenus) %>% 
  mutate(HostGenus = ifelse(HostGenus == "Deftia", "Delftia", HostGenus),
         HostGenus = ifelse(HostGenus == "Hamiltonella", "Candidatus Hamiltonella", HostGenus),
         HostGenus = ifelse(HostGenus == "Lactoccocus", "Lactococcus", HostGenus),
         HostGenus = ifelse(HostGenus == "Pseudomonad", "Pseudomonas", HostGenus),
         HostGenus = ifelse(HostGenus == "Streptomyce", "Streptomyces", HostGenus)) 

refseq.typos <- host.tax %>% 
  filter(Genus %in% refseq.bad$HostGenus) %>% 
  group_by(Phylum, Class, Order, Family, Genus) %>% 
  dplyr::count() %>% 
  arrange(Genus) %>% 
  group_by(Genus) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Keep = ifelse(Duplicates == 1, TRUE, ifelse(n == max(n), TRUE, FALSE))) %>% 
  ungroup() %>% 
  filter(Keep) %>% 
  select(-n, -Duplicates, -Keep) %>% 
  inner_join(refseq.bad, by = c("Genus" = "HostGenus")) %>% 
  mutate(HostPhylum = ifelse(is.na(Phylum), "Unassigned", as.character(Phylum)),
         HostClass = ifelse(is.na(Class), "Unassigned", as.character(Class)),
         HostOrder= ifelse(is.na(Order), "Unassigned", as.character(Order)),
         HostFamily = ifelse(is.na(Family), "Unassigned", as.character(Family)),
         HostGenus = as.character(Genus)) %>% 
  select(Genome, HostPhylum:HostGenus)
         
refseq.typos
```

There are still some genomes whose hosts were not found. 
```{r}
refseq.fail <- refseq.bad %>% 
  filter(!Genome %in% refseq.typos$Genome & !Genome %in% refseq.manual$Genome) %>% 
  select(Genome) %>% 
  mutate(HostPhylum = "Unassigned",
         HostClass = "Unassigned",
         HostOrder = "Unassigned",
         HostFamily = "Unassigned",
         HostGenus = "Unassigned") 

refseq.fail
```

Compile all the host taxonomy data frames
```{r}
refseq.all <- rbind(refseq.good,
                    refseq.typos,
                    refseq.manual,
                    refseq.fail)

refseq.all
```

Let's assign the consensus host taxonomy for each cluster. The strategy is to check each rank and see if there's only one assignment or a mix.
```{r}
clstr.h.phy <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostPhylum) %>% 
  dplyr::count() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostPhylum = ifelse(Duplicates > 1, "Mixed", as.character(HostPhylum))) %>% 
  group_by(VC, HostPhylum) %>% 
  dplyr::count() %>% 
  select(-n)

clstr.h.class <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostClass) %>% 
  dplyr::count() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostClass = ifelse(Duplicates > 1, "Mixed", as.character(HostClass))) %>% 
  group_by(VC, HostClass) %>% 
  dplyr::count() %>% 
  select(-n)

clstr.h.order <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostOrder) %>% 
  dplyr::count() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostOrder = ifelse(Duplicates > 1, "Mixed", as.character(HostOrder))) %>% 
  group_by(VC, HostOrder) %>% 
  dplyr::count() %>% 
  select(-n)

clstr.h.fam <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostFamily) %>% 
  dplyr::count() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostFamily = ifelse(Duplicates > 1, "Mixed", as.character(HostFamily))) %>% 
  group_by(VC, HostFamily) %>% 
  dplyr::count() %>% 
  select(-n)

clstr.h.gen <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostGenus) %>% 
  dplyr::count() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostGenus = ifelse(Duplicates > 1, "Mixed", as.character(HostGenus))) %>% 
  group_by(VC, HostGenus) %>% 
  dplyr::count() %>% 
  select(-n)

```

Put all the cluster info into one data frame
```{r}
clstr.master <- clstr.source %>% 
  left_join(clstr.order, by = "VC") %>% 
  left_join(clstr.family, by = "VC") %>% 
  left_join(clstr.genus, by = "VC") %>% 
  left_join(clstr.h.phy, by = "VC") %>% 
  left_join(clstr.h.class, by = "VC") %>% 
  left_join(clstr.h.order, by = "VC") %>% 
  left_join(clstr.h.fam, by = "VC") %>% 
  left_join(clstr.h.gen, by = "VC") %>% 
  mutate(Order = ifelse(is.na(Order), "Unassigned", Order)) %>% 
  mutate(Family = ifelse(is.na(Family), "Unassigned", Family)) %>% 
  mutate(Genus = ifelse(is.na(Genus), "Unassigned", Genus)) %>% 
  mutate(HostPhylum = ifelse(is.na(HostPhylum), "Unassigned", HostPhylum)) %>% 
  mutate(HostClass = ifelse(is.na(HostClass), "Unassigned", HostClass)) %>% 
  mutate(HostOrder = ifelse(is.na(HostOrder), "Unassigned", HostOrder)) %>% 
  mutate(HostFamily = ifelse(is.na(HostFamily), "Unassigned", HostFamily)) %>% 
  mutate(HostGenus = ifelse(is.na(HostGenus), "Unassigned", HostGenus))

clstr.master
```

```{r}
saveRDS(genome.master, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/genome_vc_master.RDS")
saveRDS(clstr.master, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/cluster_vc_master.RDS")
saveRDS(refseq.all, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/refseq_host_tax.RDS")
```

```{r}
ntwk <- read.table("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/c1.ntw", header = F, sep = " ", col.names = c("OTU1", "OTU2", "Score"))

nodes <- ggnet2(ntwk[,-3], mode = "fruchtermanreingold", layout.par = list(list=(niter=2000)))$data 
nodes <- nodes %>%  
  dplyr::rename("Genome" = "label")

edges <- ntwk %>% 
  mutate(Pair = paste(OTU1, OTU2, sep = ".")) %>% 
  gather(key = "Member", value = "Genome", -Pair, -Score) %>% 
  inner_join(nodes, by = "Genome")
```


Load data
```{r}
#Load orignal vConTACT2 network layout
whole.nodes <- nodes
whole.edges <- edges

#Load safe subnetwork layout
sub.nodes <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/safe_ind_subntwk_nodes.RDS")
sub.edges <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/safe_ind_subntwk_edges.RDS")

#Load safe whole network layout
safe.nodes <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/safe_ind_whole_nodes.RDS")
safe.edges <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/safe_ind_whole_edges.RDS")

#Load vContact2 cluster data
clusters.tax <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/cluster_vc_master.RDS")
clusters <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/genome_vc_master.RDS")
clusters.filt <- clusters %>% 
  select(Genome, VC, Source) %>% 
  inner_join(clusters.tax, by = "VC")

#Load Refseq host taxonomy
refseq.tax <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/refseq_host_tax.RDS")
```

Subsetting the network to only include all RefSeq nodes and the vOTUs that were used for the differential abundance analysis.
```{r}
ntwk <- whole.edges %>% 
  mutate(Source = ifelse(str_detect(Genome, "contig"), "heat", "refseq")) %>% 
  filter(Source == "refseq" | Genome %in% safe.nodes$Genome) %>% 
  group_by(Pair) %>% 
  mutate(Total = n()) %>% 
  ungroup() %>% 
  filter(Total > 1) %>% 
  select(Genome, Score, Member, Pair) %>% 
  spread(key = Member, value = Genome) %>% 
  select(OTU1, OTU2, Score)

#Generate a new layout with the filtered network
nodes <- GGally::ggnet2(ntwk[,-3], 
                mode = "fruchtermanreingold", 
                layout.par = list(list=(niter=2000))) %>% 
  .$data %>% 
  dplyr::rename("Genome" = "label") %>% 
  mutate(Source = ifelse(str_detect(Genome, "contig"), "heat", "refseq")) %>% 
  mutate(Subnetwork = ifelse(Genome %in% sub.nodes$Genome, T, F)) 

edges <- ntwk %>% 
  mutate(Pair = paste(OTU1, OTU2, sep = ".")) %>% 
  gather(key = "Member", value = "Genome", -Pair, -Score) %>% 
  inner_join(nodes, by = "Genome")  
```

Extract all the vOTUs linked to a RefSeq node
```{r}
pairs.tmp <- ntwk %>% 
  mutate(Source1 = ifelse(str_detect(OTU1, "contig"), "heat", "refseq"),
         Source2 = ifelse(str_detect(OTU2, "contig"), "heat", "refseq")) %>% 
  filter(Source1 != Source2) %>% 
  mutate(Pair = paste(OTU1, OTU2, sep = ".")) %>% 
  mutate(heatOTU = ifelse(Source1 == "heat", OTU1, OTU2),
         refseqOTU = ifelse(Source1 == "refseq", OTU1, OTU2)) %>% 
  group_by(heatOTU, refseqOTU) %>% 
  dplyr::count() %>% 
  ungroup() %>% 
  select(-n) %>% 
  dplyr::inner_join(refseq.tax, by = c("refseqOTU" = "Genome")) %>% 
  mutate(Subnetwork = ifelse(heatOTU %in% sub.nodes$Genome, T, F)) 
  
pairs.master <-  rbind(pairs.tmp %>% mutate(Set = "Whole network"),
                       pairs.tmp %>% filter(Subnetwork) %>% mutate(Set = "Subnetwork")) 
```

Save results
```{r}
saveRDS(nodes, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/refseq_ind_nodes.RDS")
saveRDS(edges, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/refseq_ind_edges.RDS")
saveRDS(pairs.master, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/refseq_ind_pairs.RDS")
```


#START HERE IF YOU JUST WANT TO REPLOT!

```{r}
#Generate palette
ntwk.palette = c(rcartocolor::carto_pal(7, "Tropic")[1:3], "gray85", rcartocolor::carto_pal(7, "Tropic")[5:7], "gray25")


#Load map
# LOAD METADATA FOR ENTIRE DATASET
metadata=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/heat_metadata.txt", delim = "\t", col_names = TRUE)
colnames(metadata)[colnames(metadata) == "Treatment"] <- "Experiment"
metadata <- metadata %>%
  mutate(Treatment = case_when(
    Temp == "Control" ~ "Field",
    Temp == "10C" ~ "Control",
    Temp == "30C" ~ "30ºC",
    Temp == "60C" ~ "60ºC", 
    Temp == "90C" ~ "90ºC")) %>%
  mutate(TempGroup = case_when(
    Treatment == "Field" ~ "Field/Control/30ºC", 
    Treatment == "Control" ~ "Field/Control/30ºC", 
    Treatment == "30ºC" ~ "Field/Control/30ºC", 
    Treatment == "60ºC" ~ "60ºC", 
    Treatment == "90ºC" ~ "90ºC")) %>%
  mutate(Virome = case_when(
    DNase == "NoDNase" ~ "NonDNase",
    DNase == "DNase" ~ "DNase"))
mapheat <- metadata %>% 
  dplyr::rename("SampleID" = "ID") 

#Load OTU table and remove singletons
#otu <- votu_tmean75
#otu <- otu[,match(map$SampleID, colnames(otu))]
#otu <- otu[rowSums(otu>0)>1,]


#Load indicator species results
a_o_indicators <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/a_o_indicators.RDS")
res.trt <- a_o_indicators  
sig.trt <- filter(res.trt, Response != "NS")

#Load safe subnetwork results

sub.nodes <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/safe_ind_subntwk_nodes.RDS")
sub.edges <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/safe_ind_subntwk_edges.RDS")

#Load safe whole network data
nodes <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/safe_ind_whole_nodes.RDS")
edges <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/safe_ind_whole_edges.RDS")

#Load links to RefSeq genomes
pairs <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/rds/refseq_ind_pairs.RDS")

#List host phyla
unique(pairs$HostPhylum)

#Format link data
pairs.filt <- pairs %>% 
  mutate(HostPhylum2 = ifelse(HostPhylum %in% c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Cyanobacteria"), as.character(HostPhylum), "Other")) %>% 
  group_by(Set, heatOTU, HostPhylum2) %>% 
  dplyr::count() %>% 
  group_by(Set, heatOTU) %>% 
  mutate(Total = n()) %>% 
  ungroup() %>% 
  mutate(HostPhylum3 = ifelse(Total == 1, HostPhylum2, "Mixed")) %>% 
  mutate(HostPhylum3 = fct_relevel(HostPhylum3, "Mixed", "Other")) %>% 
  group_by(heatOTU, HostPhylum3) %>% 
  dplyr::count() %>% 
  ungroup() %>% 
  dplyr::rename("Genome" = "heatOTU") %>% 
  select(-n)
```


```{r}
#Add indicator species annotations to network
whole.res <- nodes %>% 
  inner_join(res.trt, by = "Genome")

#Establish parameters
node.size <- 1.5
stroke.size <- 0.5

Response.ids <- whole.res$Response %>% unique()
Response.ids

#Plot
whole.ntwk.p <- whole.res %>% 
  ggplot(aes(x,y)) +
  geom_line(data = edges, aes(x, y, group = Pair), color = "black", alpha = 0.1, size = 0.2) +
  geom_point(data = filter(whole.res, Response == "NS" & !Genome %in% sub.nodes$Genome), aes(x, y, color = Response), alpha = 1, size = node.size , shape = 16) +
  geom_point(data = filter(whole.res, Response != "NS" & !Genome %in% sub.nodes$Genome), aes(x, y, color = Response), alpha = 1, size = node.size, shape = 16) +
  geom_point(data = filter(whole.res, Response == "NS" & Genome %in% sub.nodes$Genome), aes(x, y, fill = Response), alpha = 1, size = node.size , shape = 21, color = "black", stroke = stroke.size) +
  geom_point(data = filter(whole.res, Response != "NS" & Genome %in% sub.nodes$Genome), aes(x, y, fill = Response), alpha = 1, size = node.size, shape = 21, color = "black", stroke = stroke.size) +
  scale_fill_manual(values = c('#ceb0e9','#ceb0e9','mediumpurple3','gray90','gray90','greenyellow', 'greenyellow','limegreen'),
                    limits = c("A_Sensitive", "O_Sensitive", "Sensitive", "NS", "Mixed", "A_Tolerant", "O_Tolerant", "Tolerant")) +
  scale_color_manual(values = c('#ceb0e9','#ceb0e9','mediumpurple3','gray90','gray90','greenyellow', 'greenyellow','limegreen'),
                    limits = c("A_Sensitive", "O_Sensitive", "Sensitive", "NS", "Mixed", "A_Tolerant", "O_Tolerant", "Tolerant")) +
  theme(axis.text = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank(),
        axis.line = element_blank(),
        panel.grid = element_blank(),
        panel.background = element_rect(fill = "white"),
        legend.position = "none")

whole.ntwk.p

ggsave("wholentwk.pdf", width = 200, height = 175, units = "mm", dpi = 500)

```
```{r}
#Add indicator species annotations and host taxonomy to subnetwork
sub.res <- sub.nodes %>% 
  inner_join(res.trt, by = "Genome") %>% 
  left_join(pairs.filt, by = "Genome") %>% 
  mutate(HostPhylum = ifelse(is.na(HostPhylum3), "Unassigned", as.character(HostPhylum3)))

#Plot
subntwk.p <- sub.res %>% 
  ggplot() +
  geom_line(data = sub.edges, aes(x, y, group = Pair), color = "black", alpha = 0.25, size = 0.1) +
  geom_point(data = filter(sub.res, Response == "NS"), aes(x, y, color = Response), alpha = 1, size = 5 , shape = 16) +
  geom_point(data = filter(sub.res, Response != "NS"), aes(x, y, color = Response), alpha = 1, size = 5, shape = 16) +
  geom_label(data = filter(sub.res, HostPhylum == "Proteobacteria"), aes(x, y), label = "P", size = 5, alpha = 0.2, shape = 1) +
  scale_color_manual(values = c('#ceb0e9','#ceb0e9','mediumpurple3','gray90','gray90','greenyellow', 'greenyellow','limegreen'),
                    limits = c("A_Sensitive", "O_Sensitive", "Sensitive", "NS", "Mixed", "A_Tolerant", "O_Tolerant", "Tolerant")) +
  theme(axis.text = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank(),
        axis.line = element_blank(),
        panel.grid = element_blank(),
        panel.background = element_rect(fill = "white"),
        legend.position = "none")

subntwk.p

ggsave("subntwk.pdf", width = 200, height = 175, units = "mm", dpi = 500)

```

